QuickNoteVault.API/QuickNoteVault.BLL/Services/NoteService.cs

```diff
@@ -71,13 +71,56 @@ public async Task AddAsync(NoteModel noteModel)
 
     public async Task UpdateAsync(NoteModel noteModel)
     {
-        await Task.CompletedTask;
-        throw new NotImplementedException();
+        if (noteModel == null || noteModel.Content == null)
+        {
+            throw new NoteNotFoundException();
+        }
+
+        if (noteModel.UserId <= 0)
+        {
```

Comment:
Remove ` if (noteModel.UserId <= 0)` block. Rename `existingNote` to `oldNoteEntity`. After getting old note add `if (oldNoteEntity.UserId != noteModel.UserId)
{
    throw new UserNotFoundException();
}`. Rename `noteEntity` to `newNoteEntity`. After mapping replace 3 line of code with `newNoteEntity.Title = newNoteEntity.Title.Trim();
newNoteEntity.CreatedAt = oldNoteEntity.CreatedAt;
newNoteEntity.ModifiedAt = DateTime.UtcNow;`. Replace `_notes.Update(existingNote);` with `_notes.Update(newNoteEntity);`